bluepysnap/edges.py

```diff
@@ -27,8 +26,353 @@
+            elif d[index] != value:
+                raise BluepySnapError("Same property with different "
+                                      "dtype. {}: {}!= {}".format(index, value, d[index]))
+
+        res = dict()
+        for pop in self.populations():
+            for varname, dtype in pop.property_dtypes.iteritems():
+                _update(res, varname, dtype)
+        return pd.Series(res)
+
+    def keys(self):
+        """Returns iterator on the population names.
+
+        Made to simulate the behavior of a dict.keys().
+        """
+        return (name for name in self.population_names)
+
+    def values(self):
+        """Returns iterator on the EdgePopulations.
+
+        Made to simulate the behavior of a dict.values().
+        """
+        return (self[name] for name in self.population_names)
+
+    def items(self):
+        """Returns iterator on the tuples (population name, EdgePopulations).
+
+        Made to simulate the behavior of a dict.items().
+        """
+        return ((name, self[name]) for name in self.population_names)
+
+    # helper renaming
+    names = keys
+    populations = values
+
+    def __getitem__(self, population_name):
+        """Access the EdgePopulation corresponding to the population 'population_name'."""
+        try:
+            return self._populations[population_name]
+        except KeyError:
+            raise BluepySnapError("{} not a edge population.".format(population_name))
+
+    def __iter__(self):
+        """Allows iteration over the different EdgePopulation."""
+        return iter(self.names())
+
+    @cached_property
+    def size(self):
+        """Total number of edges inside the circuit."""
+        return sum(pop.size for pop in self.populations())
+
+    @cached_property
+    def property_names(self):
+        """Returns all the properties present inside the circuit."""
+        return set(prop for pop in self.populations() for prop in pop.property_names)
+
+    def _get_ids_from_pop(self, fun_to_apply, returned_ids_cls):
+        """Get CircuitIds of class 'returned_ids_cls' for all populations using 'fun_to_apply'.
+
+        Args:
+            fun_to_apply (function): A function that returns the list of IDs for each population
+                and the population containing these IDs.
+            returned_ids_cls (CircuitNodeIds/CircuitEdgeIds): the class for the CircuitIds.
+
+        Returns:
+            CircuitNodeIds/CircuitEdgeIds: containing the IDs and the populations.
+        """
+        ids = np.empty((0,), dtype=np.int64)
+        str_type = "<U{}".format(max(len(pop) for pop in self.population_names))
+        populations = np.empty((0,), dtype=str_type)
+        for pop in self.populations():
+            pop_ids, name_ids = fun_to_apply(pop)
+            pops = np.full_like(pop_ids, fill_value=name_ids, dtype=str_type)
+            ids = np.concatenate([ids, pop_ids])
+            populations = np.concatenate([populations, pops])
```

Comment:
It's probably better to concatenate only once at the end:

```suggestion
        populations = list()
        ids = list()
        for pop in self.populations():
            pop_ids, name_ids = fun_to_apply(pop)
            populations.append(np.full_like(pop_ids, fill_value=name_ids, dtype=str_type))
            ids.append(name_ids)
        ids = np.concatenate(ids)
        populations = np.concatenate(populations)
```

Reply:
Totally agree